app/products/calls/connection/websocket_client.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {EventEmitter} from 'events';
+
+import {encode} from '@msgpack/msgpack/dist';
+
+import Calls from '@constants/calls';
+import DatabaseManager from '@database/manager';
+import {getCommonSystemValues} from '@queries/servers/system';
+
+export default class WebSocketClient extends EventEmitter {
```

Comment:
We don't really need it (this was mostly ported from the plugin side) but it's been convenient to have it so far since we didn't want to touch the existing one as we required some special handling/functionality (e.g. sending binary messages). I can definitely see us converging into a single client/connection at some point but not a huge priority at this time.

Reply:
I kind of disagree, is yet anither connection that needs to be kept alive consuming further resources

Reply:
I assume you are disagreeing on "not a huge priority at this time" because otherwise I think we all agree it would be more efficient to have a single connection :) 

My point here is that this PR is about porting existing functionality from v1 to v2 and this is how Calls has been working from the beginning on both sides (web and mobile). I understand it's not optimal but at the same time it's not something so impacting as to become P0 as we are talking about a single connection overhead that would only be open when a user is in a call and there can only be one at any given time.

Now, coming to next steps, I created a [card](https://community.mattermost.com/boards/workspace/545qxftftbdpu8amtqquwhudie/bceotfube8jfc7bbsu4yj9hdzjr/vkk6nrmzc778fjmfuhd4oyrzecc/c1uk7gusrktdcfp6qfqy711rinw) to address this sometime before GA. Please let me know if this sounds reasonable to you.

Reply:
Yeah it is about "at this time" but if the connections is only created while the call is joined, then I'm ok to leave it like this